batch2/batch/k8s.py

```diff
@@ -0,0 +1,112 @@
+import logging
+
+import kubernetes as kube
+import prometheus_client as pc
+
+from .blocking_to_async import blocking_to_async
+
+log = logging.getLogger('batch.k8s')
+
+
+class EmptyContextManager:
+    def __enter__(self):
+        return None
+
+    def __exit__(self, type, value, callback):
+        return None
+
+
+class NoSummary:
+    EMPTY_CONTEXT_MANAGER = EmptyContextManager()
+
+    def time(self):  # pylint: disable=R0201
+        return NoSummary.EMPTY_CONTEXT_MANAGER
+
+
+class K8s:
```

Comment:
I was a bit surprised by this.  Looks like you only call read_secret.  we should pare this down.